css-view-transitions-2/Overview.bs

```diff
@@ -0,0 +1,373 @@
+
+		1. If |newDocument| is given, then [=setup outbound cross-document view transition=] given |oldDocument| and |newDocument|.
+	</div>
+
+	<div algorithm="monkey patch to render blocking">
+		Run the following step at the end of [=unblock rendering=]:
+
+		1. [=reveal document|reveal=] |document|.
+
+		Reword the text in the [=render-blocked=] definition, to call [=reveal document=] |document| if the [=implementation-defined=] timeout value has been exceeded.
+	</div>
+
+	<div algorithm="monkey patch to reactivation">
+		Run the following step in [=Document/reactivate=], before step 4 (querying for [=page showing=]):
+
+		1. [=reveal document|reveal=] |document|.
+	</div>
+
+	<div algorithm="page reveal">
+	To <dfn>reveal {{Document}}</dfn> |document|:
+		1. Assert: |document|'s [=page showing=] is false.
+
+		1. If |document| is [=render-blocked=] then return.
+
+		1. Fire a new event named <code>reveal</code> on |document|'s [=relevant global object=],
+			using {{PageRevealEvent}}.
+
+		1. Let |transition| be the result of getting the [=inbound cross-document view-transition=] for |document|.
+
+		1. If |transition| is not null, then [=activate view transition|activate=] |transition|.
+	</div>
+
+## Setting up and activating the cross-document view transition
+
+	<div algorithm>
+		To <dfn>setup outbound cross-document view transition</dfn> given a {{Document}} |oldDocument|
+		and a {{Document}} |newDocument|:
+
+		1. Assert: this is called when |oldDocument| is [=unloading=].
+
+		1. If |oldDocument|'s [=active view transition=] is not null,
+			then [=skip the view transition|skip=] |oldDocument|'s [=active view transition=]
+			with an "{{AbortError}}" {{DOMException}} in |oldDocument|'s [=relevant Realm=].
+
+		1. If |oldDocument|'s [=environment settings object/origin=] is not [=same origin=] as
+			|newDocument|'s [=environment settings object/origin=] then return.
+
+		1. If |newDocument| [=was created via cross-origin redirects=] is true
+			and |newDocument|'s [=latest entry=] is null, then return.
+
+			Note: A document with a non-null [=latest entry=]
+			is being [=Document/reactivated=], in which case we don't need to check for cross-origin redirects.
+
+		1. If |oldDocument| does not [=opt in to cross-document view transitions=], then return.
+
+			Note: we don't know yet if |newDocument| has opted in, as it might not be parsed yet.
+			We check the opt-in for |newDocument| when it is [=reveal document|revealed=].
+
+		1. Let |transition| be a new {{ViewTransition}} object in |oldDocument|'s [=relevant Realm=].
+
+		1. [=Capture the old state=] for |transition|. If this fails, return.
+
+			Note: implementations might need to [=queue a task=] in order to [=capture the old state=].
+			However, this should not have web-observable side-effects.
+
+		1. [=Queue a global task=] on the [=DOM manipulation task source=],
```

Comment:
Not sure about a task here referencing the new Document here? It feels like we should stash the old doc's transition object somewhere and then as a part of creating the new Document create the VT object which copies over state from it (as in the step below).

We basically want an async phase between deciding that the navigation will be committed but before the unload steps are run. This global task should unblock running the unload steps.

Reply:
Revised this.